azurerm/internal/services/datafactory/data_factory_linked_service_data_lake_storage_gen2_resource.go

```diff
@@ -58,21 +58,27 @@ func resourceArmDataFactoryLinkedServiceDataLakeStorageGen2() *schema.Resource {
 				ValidateFunc: validation.IsURLWithHTTPS,
 			},
 
+			"use_managed_identity": {
+				Type:     schema.TypeBool,
+				Optional: true,
```

Comment:
We'll want to add `ConflictsWith` here too to make sure we don't specify it with the other service principal attributes.

Reply:
And to round it all out we should throw in `AtLeastOneOf` with all of these attributes to make sure at least one of them gets specified when creating this resource.